node/popn.go

```diff
@@ -1024,27 +1120,61 @@ func (nd *node) connPeers() []peer.ID {
 	return out
 }
 
-// importAddress from a hex encoded private key to use as default on the exchange instead of
+// importPrivateKey from a hex encoded private key to use as default on the exchange instead of
 // the auto generated one. This is mostly for development and will be reworked into a nicer command
 // eventually
-func (nd *node) importAddress(pk string) {
+func (nd *node) importPrivateKey(ctx context.Context, pk string) error {
 	var iki wallet.KeyInfo
+
 	data, err := hex.DecodeString(pk)
 	if err != nil {
-		log.Error().Err(err).Msg("hex.DecodeString(opts.PrivKey)")
+		return fmt.Errorf("failed to decode key: %v", err)
+	}
+
+	err = iki.FromBytes(data)
+	if err != nil {
+		return fmt.Errorf("failed to decode keyInfo: %v", err)
 	}
-	if err := json.Unmarshal(data, &iki); err != nil {
-		log.Error().Err(err).Msg("json.Unmarshal(PrivKey)")
+
+	addr, err := nd.exch.Wallet().ImportKey(ctx, &iki)
+	if err != nil {
+		return fmt.Errorf("failed to import key: %v", err)
 	}
 
-	addr, err := nd.exch.Wallet().ImportKey(context.TODO(), &iki)
+	err = nd.exch.Wallet().SetDefaultAddress(addr)
 	if err != nil {
-		log.Error().Err(err).Msg("Wallet.ImportKey")
-	} else {
-		fmt.Printf("==> Imported private key for %s.\n", addr.String())
-		err := nd.exch.Wallet().SetDefaultAddress(addr)
-		if err != nil {
-			log.Error().Err(err).Msg("Wallet.SetDefaultAddress")
-		}
+		return fmt.Errorf("failed to set default address: %v", err)
 	}
+
+	return nil
+}
+
+// importPrivateKey from a hex encoded private key to use as default on the exchange instead of
```

Comment:
Seems like copy pasted comment from above. If you're not reusing this `exportPrivateKey` method I think the logic can all be in the `WalletExport`.

Reply:
I'm actually using it in the popn walletExport : https://github.com/myelnet/pop/blob/ag/improve-default-fil-key-ux/node/popn.go#L515

Reply:
Yeah I just meant why have a separate `exportPrivateKey` method and not have all the logic in `WalletExport`?